### PR TITLE
Add calculate_rate for NetworkIn/Out, DiskRead/WriteBytes and DiskRead/WriteOps

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
This PR is to try solve https://github.com/elastic/kibana/issues/42687 in metricbeat side.
In ec2 metricset, adding a config parameter called `calculate_rate` to calculate rates for 6 metrics:

network.in.bytes_per_sec = network.in.bytes / 300 or network.in.bytes / 60
network.out.bytes_per_sec = network.out.bytes / 300 or network.out.bytes / 60
diskio.read.bytes_per_sec = diskio.read.bytes / 300 or diskio.read.bytes / 60
diskio.write.bytes_per_sec = diskio.write.bytes / 300 or diskio.write.bytes / 60
diskio.read.ops_per_sec = diskio.read.ops / 300 or diskio.read.ops / 60
diskio.write.ops_per_sec = diskio.write.ops / 300 or diskio.write.ops / 60

Default for `calculate_rate` is false for now.

### CODE REVIEW

**Comment by kaiyan-sheng** [comment]
If detailed monitoring is enabled, then these metrics will be sent to Cloudwatch every 1 min instead of every 5min. The rate calculated here is for each data point, which has nothing to do with the collection period that user sets in Metricbeat aws module. Therefore this calculation doesn't seem to be able to replace by adding collection period into each event.  @jsoriano  What do you think?

File: x-pack/metricbeat/module/aws/_meta/config.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@
     - ec2
   regions:
     - us-east-1
+  calculate_rate: false
```
Comment (jsoriano) [inline x-pack/metricbeat/module/aws/_meta/config.yml]: I would always calculate it, the other value is difficult to consume without more context, and adding metrics is non-breaking in principle.

File: x-pack/metricbeat/module/aws/ec2/_meta/docs.asciidoc [hunk 2]
```diff
@@ -47,6 +47,20 @@ The aws ec2 metricset comes with a predefined dashboard. For example:
...
+|===
+|Rate Metric Name | Equation | Equation with Detailed Monitoring
+|network.in.bytes_per_sec | network.in.bytes / 300 | network.in.bytes / 60
+|network.out.bytes_per_sec | network.out.bytes / 300 | network.out.bytes / 60
+|diskio.read.bytes_per_sec | diskio.read.bytes / 300 | diskio.read.bytes / 60
+|diskio.write.bytes_per_sec | diskio.write.bytes / 300 | diskio.write.bytes / 60
+|diskio.read.ops_per_sec | diskio.read.ops / 300 | diskio.read.ops / 60
+|diskio.write.ops_per_sec | diskio.write.ops / 300 | diskio.write.ops / 60
+|===
+Default calculate_rate is false.
```
Comment (jsoriano) [inline x-pack/metricbeat/module/aws/ec2/_meta/docs.asciidoc]: I'd remove this explanation and just always calculate the rate for affected metrics.

File: x-pack/metricbeat/module/aws/ec2/_meta/data.json [hunk 3]
```diff
@@ -47,31 +51,37 @@
             },
             "network": {
                 "in": {
-                    "bytes": 85.2,
-                    "packets": 1.4
+                    "bytes": 1297480,
+                    "bytes_per_sec": 4324.933333333333,
+                    "packets": 2734.8
```
Comment (jsoriano) [inline x-pack/metricbeat/module/aws/ec2/_meta/data.json]: Should we also calculate `packets_per_second`?

File: x-pack/metricbeat/module/aws/ec2/ec2.go [hunk 4]
```diff
@@ -239,6 +261,49 @@ func (m *MetricSet) createCloudWatchEvents(getMetricDataResults []cloudwatch.Met
...
+	if err == nil && networkOut != nil {
+		diskReadOpsRate := diskReadOps.(float64) / period
+		resultMetricsetFields.Put("diskio.read.count_per_sec", diskReadOpsRate)
+	}
+
+	diskWriteOps, err := resultMetricsetFields.GetValue("diskio.write.count")
+	if err == nil && networkOut != nil {
+		diskWriteOpsRate := diskWriteOps.(float64) / period
+		resultMetricsetFields.Put("diskio.write.count_per_sec", diskWriteOpsRate)
+	}
```
Comment (jsoriano) [inline x-pack/metricbeat/module/aws/ec2/ec2.go]: A lot of repeated code, could we put it in a loop or a function?

File: x-pack/metricbeat/module/aws/_meta/config.yml (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/_meta/config.yml]: Sounds good, I will remove the flag then!

File: x-pack/metricbeat/module/aws/ec2/_meta/data.json (hunk 3 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/ec2/_meta/data.json]: Yes, ui team also brought this up. I will add it in next PR.

File: x-pack/metricbeat/module/aws/ec2/ec2.go (hunk 4 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/ec2/ec2.go]: yep, done 👍

File: x-pack/metricbeat/module/aws/ec2/ec2.go [hunk 5]
```diff
@@ -239,6 +249,25 @@ func (m *MetricSet) createCloudWatchEvents(getMetricDataResults []cloudwatch.Met
...
 
+func calculateRate(resultMetricsetFields common.MapStr, monitoringState string) {
+	var period = 300.0
+	if monitoringState != "disabled" {
+		period = 60.0
+	}
+
+	metricList := []string{"network.in.bytes", "network.out.bytes",
+		"network.in.packets", "network.out.packets", "diskio.read.bytes",
+		"diskio.write.bytes", "diskio.read.count", "diskio.write.count"}
```
Comment (jsoriano) [inline x-pack/metricbeat/module/aws/ec2/ec2.go]: Nit. For readability this could be unwrapped, one metric per line.

**Pull request merged**